### PR TITLE
Avoid extra queries for connections

### PR DESCRIPTION
**Pull request by DmitryTsepelev**
Instead of re–assembling connection objects when cache is restored, we can just cache the object itself: in this case we don't care what fields were not empty before caching (most of them are lazy). We don't have the direct access to the connection objects, because relations are wrapped to connections lazily (after resolve), so we have to use tracer to catch them.

### CODE REVIEW

File: lib/graphql/fragment_cache/cacher.rb [hunk 1]
```diff
@@ -10,9 +10,7 @@ class << self
         def call(query)
           return unless query.context.fragments?
 
-          final_value = query.context.namespace(:interpreter)[:runtime].final_value
-
-          query.context.fragments.each { _1.persist(final_value) }
+          query.context.fragments.each(&:persist)
```
Comment (DmitryTsepelev) [inline lib/graphql/fragment_cache/cacher.rb:13]: Fragment decides where to find value to cache

File: lib/graphql/fragment_cache/fragment.rb [hunk 2]
```diff
@@ -6,22 +6,24 @@ module GraphQL
...
       def read
         FragmentCache.cache_store.read(cache_key)
       end
 
-      def persist(final_value)
-        value = raw_connection || resolve(final_value)
+      def persist
+        # Connections are not available from the runtime object, so
+        # we rely on Schema::Tracer to save it for us
+        value = resolved_value || resolve_from_runtime
```
Comment (DmitryTsepelev) [inline lib/graphql/fragment_cache/fragment.rb:26]: If `resolved_value` is set—fragment prefers it

File: lib/graphql/fragment_cache/schema/tracer.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+          def connection_to_cache?(key, data)
+            key == "execute_field" && data[:field].connection?
+          end
+
+          def verify_connections!(context)
+            return if context.schema.new_connections?
+
+            raise StandardError,
+              "GraphQL::Pagination::Connections should be enabled for connection caching"
```
Comment (palkan) [inline lib/graphql/fragment_cache/schema/tracer.rb:32]: Can we add a link to the docs/guide on what is it and how to enable it?

Comment (DmitryTsepelev) [reply lib/graphql/fragment_cache/schema/tracer.rb:32]: There is a [mention](https://github.com/DmitryTsepelev/graphql-ruby-fragment_cache/blame/master/README.md#L64) in the README

**Pull request merged**